src/cf.data.pre

```diff
@@ -1043,6 +1043,21 @@ DOC_START
 	  # NOTE 2: IPv6 protocol does not contain ARP. MAC/EUI is either
 	  # encoded directly in the IPv6 address or not available.
 
+	acl aclname clientside_mark mark[/mask] ...
+	  # matches CONNMARK of an accepted connection [fast]
+	  #
+	  # mark and mask are unsigned integers (hex, octal, or decimal).
+	  # If multiple marks are given, then the ACL matches if at least
+	  # one mark matches.
+	  #
+	  # Uses netfilter-conntrack library.
+	  # Requires building Squid with --enable-linux-netfilter.
+	  #
+	  # The client, various intermediaries, and Squid itself may set
+	  # CONNMARK at various times. The last CONNMARK set wins. However,
+	  # this ACL only checks the mark present on a freshly accepted
+	  # connection, before clientside_mark is checked.
```

Comment:
That last sentence is incorrect. This ACL is based on the Comm::Connection stored mark. Which may have been updated already on a persistent connection by a previous HTTP request use of clientside_mark, or by the directive using the ACL happening after clientside_mark for the current transaction. eg http_access uses the previous transaction (or client original) marking, whereas logging for the same transaction uses the clientside_mark changed value.

The most that can be said is that this ACL checks for the most recent CONNMARK used on the client connection.

Reply:
> That last sentence is incorrect.

Agreed. I probably missed setSockNfmark() calls when drafting it. Sorry.

> The most that can be said is that this ACL checks for the most recent CONNMARK used on the client connection.

That does not tell the admin what they need to know IMO: It is important to warn the admin that the mark they are testing at time X may not match the actual/real connection mark at that time. How about the following?

```
	  # The client, various intermediaries, and Squid itself may set
	  # CONNMARK at various times. The last CONNMARK set wins. This ACL
	  # checks the mark present on an accepted connection or set by
	  # Squid afterwards, depending on the ACL check timing. This ACL
	  # effectively ignores any mark set by other agents after Squid has
	  # accepted the connection.
```